tests/state/conftest.py

```diff
@@ -42,20 +42,39 @@
     get_crosslink_notaries,
 )
 
-import beacon_chain.utils.bls as bls
+import beacon_chain.utils.bls
 from beacon_chain.utils.blake import (
     blake,
 )
 from beacon_chain.utils.simpleserialize import (
     serialize,
 )
 
+bls = beacon_chain.utils.bls
 
 DEFAULT_SHUFFLING_SEED = b'\35'*32
 DEFAULT_RANDAO = b'\45'*32
 DEFAULT_NUM_VALIDATORS = 40
 
 
+def bls_verify_mock(m, pub, sig):
+    return True
```

Comment:
Suggestion: Mock

- `sign` to return `(m, privtopub(k))`
- `aggregate_pubs` to return `pubs`
- `aggregtae_sigs` to return `(m, [pub1, pub2, ...])`
- `verify(m, pub, sig)` to check `sig == (m, pub)`
- `privtopub` to return `priv` (or maybe some hash of `priv`)

Then the mocked checks would still check that the message has not been tampered with and that the right keys have signed it.

Reply:
Great suggestion. I'll do a timing analysis and check it out

Reply:
@jannikluhn Having sigs of this artificial type breaks serialization of some objects because they assume sigs are of the type `['int256']`. I can go in an mock various components of objects to fix this but it seems a bit over engineered for our current purposes.

I'm currently leaning toward a simple bls mock that assumes no messages have been tampered with and correct keys have signed, at least for the time being.

Thoughts?

[getting errors like the following due to serialization problems]:
```base
tests/state/conftest.py:403: in mock_make_child
    block.sign(keymap[crystallized_state.active_validators[proposer].pubkey])
beacon_chain/state/block.py:49: in sign
    self.sig = list(sign(serialize(self), key))
beacon_chain/utils/simpleserialize.py:22: in serialize
    [serialize(getattr(val, k), typ.fields[k]) for k in sorted(typ.fields.keys())]
beacon_chain/utils/simpleserialize.py:22: in <listcomp>
    [serialize(getattr(val, k), typ.fields[k]) for k in sorted(typ.fields.keys())]
beacon_chain/utils/simpleserialize.py:18: in serialize
    sub = b''.join([serialize(x, typ[0]) for x in val])
beacon_chain/utils/simpleserialize.py:18: in <listcomp>
    sub = b''.join([serialize(x, typ[0]) for x in val])
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

val = b'\x00\x00\x01<\x00\x00\x00@\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x...0\xe4\x03\xff?\x90\xb5\xe5\x02\xfc\x99\xeeB=l\xea\xd0].\xe1dX\x86z\xf6\xe1\xf9\x11(79\x02v[\x1e\x9a\xed\x8bc\x05l\xfa7'
typ = 'int256'

    def serialize(val, typ=None):
        if typ is None and hasattr(val, 'fields'):
            typ = type(val)
        if typ in ('hash32', 'address'):
            assert len(val) == 20 if typ == 'address' else 32
            return val
        elif isinstance(typ, str) and typ[:3] == 'int':
            length = int(typ[3:])
            assert length % 8 == 0
>           return val.to_bytes(length // 8, 'big')
E           AttributeError: 'bytes' object has no attribute 'to_bytes'

beacon_chain/utils/simpleserialize.py:13: AttributeError
```

Reply:
Hm, I thought it was quite straightforward, but I didn't think about serialization. In that case I agree that it's not worth the number of hacks required to make it work (the gains are quite minimal anyways).